stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -1589,18 +1606,54 @@ internal extension _ExecutionContext {
 @inlinable
 public func withDevice<R>(_ kind: DeviceKind,
                           perform body: () throws -> R) rethrows -> R {
-  _ExecutionContext.global.pushDeviceScope(kind)
+  // internalConsistencyCheck(false, "pushing device scope \(kind).")
+  _ExecutionContext.global.pushDeviceScope(kind, 0)
   let result = try body()
   _ExecutionContext.global.popDeviceScope()
   return result
 }
 
+@_silgen_name("_swift_tfc_PushDeviceScope")
+@usableFromInline @inline(never)
+internal func pushDeviceScope(_ kind: DeviceKind, _ index: UInt32) {
+  debugLog("pushing device scope (\(kind), \(index)).")
+  // internalConsistencyCheck(false, "pushing device scope (\(kind), \(index)).")
+  _ExecutionContext.global.pushDeviceScope(kind, index)
+}
+
+@_silgen_name("_swift_tfc_PopDeviceScope")
+@usableFromInline @inline(never)
+internal func popDeviceScope() {
+  debugLog("popping device scope.")
+  _ExecutionContext.global.popDeviceScope()
+}
+
+/// Same as above, but with device index support.
+// TODO: merge with `withDevice` above.
+//
+// @inline(never)
+//
+// this could get inlined before partition pass starts, in which case we
+// generate a partitionable func at the caller.
+
+@inline(never)
+// @inlinable // @inline(__always)
+// @_silgen_name("_swift_tfc_withDevice")
+public func withDevice<R>(_ kind: DeviceKind, _ index: UInt32,
+                          perform body: () throws -> R) rethrows -> R {
+  // internalConsistencyCheck(false, "pushing device scope (\(kind), \(index)).")
```

Comment:
Delete this as well? This duplicates line 1620 in `pushDeviceScope`'s body.

Reply:
Done. The changes in this file are mostly temporary, and have been reverted because the proper PR form Parker has now landed.